text/0019-milestone-payload/0019-milestone-payload.md

```diff
@@ -0,0 +1,136 @@
+    <td>Signatures Count</td>
+    <td>uint8</td>
+    <td>Number of signature entries. The number must match the field <code>Keys Count</code>.</td>
+  </tr>
+  <tr>
+    <td>Signatures</td>
+    <td>Array&lt;ByteArray[64]&gt;</td>
+    <td>An array of signatures signing the serialized <i>Milestone Essence</i>. The signatures must be in the same order as the specified public keys.</td>
+  </tr>
+</table>
+
+## Generation
+
+- Generate a new _Milestone Essence_ corresponding to the Coordinator milestone.
+- Transmit the serialized _Milestone Essence_ to the corresponding number of signature service providers.
+  - The signature provider service will sign the received serialized bytes as-is.
+  - The signature provider will serialize the signature bytes and return them to the Coordinator.
+- Fill the `Signatures` field of the milestone payload with the received signature bytes.
+- Generate a *Message* as defined in [RFC-0017 (draft)](https://github.com/GalRogozinski/protocol-rfcs/blob/message/text/0017-message/0017-message.md) using the same `Parent1` and `Parent2` for the created _Milestone Payload_.
+
+## Syntactical validation
+
+- `Parent1` and `Parent2` of the payload must match `Parent1` and `Parent2` of the encapsulating _Message_.
+- `Keys Count` must be at least the _Signature Threshold_ and at most the number of _Applicable Public Keys_ for the current milestone index.
+- Public keys:
+  - The provided keys must form a subset of the _Applicable Public Keys_ for the current milestone index.
+  - The keys must be in lexicographical order.
+- `Signatures Count` must match the amount of public keys. 
+- All `Signatures` must be valid.
+- Given the type and length information, the _Milestone Payload_ must consume the entire byte array of the `Payload` field of the _Message_.
+
+# Rationale and alternatives
+
+Instead of going with EdDSA we could have chosen ECDSA. Both algorithms are well supported and widespread. However, signing with ECDSA requires fresh randomness while EdDSA does not. Moreover, we could have used a commit-reveal mechanism to update and commit the Coordinator public key at each next milestone. This method would make a quantum-based attack aimed at breaking the "current" Coordinator private-key more difficult. On the other hand, key management as well as verification of the milestone chain would become more complex.
+
+# Unresolved questions
+
+- Forcing matching `Parent1`, `Parent2` in the _Milestone Payload_ and its _Message_ makes it impossible to reattach the same payload at different positions in the Tangle. While this does not prevent reattachments in general (a different, valid `Nonce`, for example would lead to a new Message ID), this still simplifies milestone processing. However, it violates a clean separation of payload and message. As such, it might still be desirable to slightly complicate the milestone processing in order to allow any reattachments of _Milestone Payloads_ by not validating the parents.
```

Comment:
"in order to allow"?

we must not allow reattachments

Reply:
It seems to be practically infeasible/impossible to completely prevent reattachments. As such, it is much easier and cleaner to consider the Milestone as a "virtual marker" located at the corresponding position than an actual message location. I've adapted the RFC to clarify this aspect.